libraries/FunctionalInterrupt/examples/Functional/Functional.ino

```diff
@@ -0,0 +1,75 @@
+// Hint: On ESP8266, this no longer depends on FunctionalInterrupt, but is rather a feature of
+// the standard API in Arduino.h
+
+//#include <FunctionalInterrupt.h>
+
+#ifndef IRAM_ATTR
+#define IRAM_ATTR ICACHE_RAM_ATTR
+#endif
+
+#if defined(ESP32)
+#define BUTTON1 16
+#define BUTTON2 17
+#elif defined(ARDUINO_ESP8266_WEMOS_D1MINI)
+#define BUTTON1 D4
+#define BUTTON2 D3
+#else
+#define BUTTON1 2
+#define BUTTON2 0
+#endif
+
+class Button {
+  public:
+    Button(uint8_t reqPin) : PIN(reqPin) {
+      pinMode(PIN, INPUT_PULLUP);
+      //attachInterruptArg(PIN, reinterpret_cast<void(*)(void*)>(&buttonIsr_static), this, FALLING);
+      attachInterrupt(PIN, std::bind(&Button::buttonIsr, this), FALLING);
+    };
+    ~Button() {
+      detachInterrupt(PIN);
+    }
+
+    void IRAM_ATTR buttonIsr() {
+      numberKeyPresses += 1;
+      pressed = true;
+    }
+
+    static void IRAM_ATTR buttonIsr_static(Button* const self) {
+      self->buttonIsr();
+    }
+
+    uint32_t checkPressed() {
```

Comment:
checkClearPressed(), to be clear about what happens (I was going to suggest const method, then I realized that this isn't just a check, but also a clear of a member)

Reply:
Opted for checkResetPressed()

Reply:
Opted for testResetPressed()